### ISSUE TITLE
Avoid saving logits

### ISSUE THREAD

**Issue by Heisenburger2020** [yongliang-wu/DFT#17]
Can i do "loss = loss * torch.exp(-loss.detach())" rather than "loss = loss * torch.softmax(shift_logits, dim=-1).gather(1, shift_labels.unsqueeze(-1)).squeeze(-1).detach()" to save cuda memory, since logits can be very large
